File: web/src/api/system/dict.js
import request from '@/utils/request'

// 创建
export function createDict(data) {
  return request({
    url: '/api/dict/create',
    method: 'post',
    data
  })
}

// 列表
export function listDict(params) {
  return request({
    url: '/api/dict/list',
    method: 'get',
    params
  })
}

// 更新
export function putDict(data) {
  return request({
    url: '/api/dict/put',
    method: 'put',
    data
  })
}

// 删除
export function deleteDict(data) {
  return request({
    url: '/api/dict/delete',
    method: 'delete',
    data
  })
}

// 批量删除
export function removeDict(data) {
  return request({
    url: '/api/dict/remove',
    method: 'delete',
    data
  })
}

File: server/app/router/system/dict.go
package system

import (
	"github.com/gin-gonic/gin"
	"server/app/api/system"
)

// InitExampleRouter 用户案例模块
func InitDictRouter(r *gin.RouterGroup) gin.IRouter {
	dictRouter := system.NewDictApi()
	router := r.Group("/dict")
	// 开启jwt认证中间件
	//router.Use(global.AuthMiddleware.MiddlewareFunc())
	// 开启casbin鉴权中间件
	//router.Use(middleware.CasbinMiddleware())

	{
		router.POST("/create", dictRouter.PostDict)        // 创建
		router.GET("/list", dictRouter.GetDictList)        // 列表
		router.PUT("/put", dictRouter.PutDict)             // 更新
		router.DELETE("/delete", dictRouter.DeleteDict)    // 删除
		router.DELETE("/remove", dictRouter.DeleteDictAll) // 全部删除
	}

	return r
}
